azurerm/internal/services/compute/shared_schema.go

```diff
@@ -0,0 +1,433 @@
+package compute
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func bootDiagnosticsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				// TODO: should this be `storage_account_endpoint`?
```

Comment:
since this is the URI of the Storage Account rather than the Name of it (e.g. `https://account1.blob.core.windows.net` - it could potentially be renamed to `storage_account_endpoint`; but I think that's one we'll want to fix up once the new VM & VMSS resources are merged but before they ship - I'll update this TODO and fix this once that's merged)